pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -476,60 +499,88 @@ func (h *handler) OnChange(obj runtime.Object) (runtime.Object, error) {
 		return obj, generic.ErrSkip
 	}
 
-	newObj, err := h.run(infra, true)
-	if newObj == nil {
-		newObj = obj
+	create := infra.meta.GetDeletionTimestamp().IsZero()
+
+	if !create {
+		// do not recreate jobs if about to delete
+		if cond := getCondition(infra.data, deleteJobConditionType); cond != nil && cond.Status() == "True" {
+			return obj, nil
+		}
+	}
+
+	state, failedCreate, err := h.run(infra, create)
```

Comment:
We shouldn't run delete jobs ever from the OnChange handler, that should solely be in the OnRemove handler